src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,304 @@
+                    + "| | | | | | | |/ / _ \\\n"
+                    + "| |_| | |_| |   <  __/\n"
+                    + "|____/ \\__,_|_|\\_\\___|\n";
+        System.out.println(HORIZONTAL_LINE+ "\n" + logo);
+    }
+    public static void greet() {
+        String greeting = "Hello! I'm duke.Duke\n"
+                        + "What can I do for you?\n";
+        System.out.println(HORIZONTAL_LINE + "\n" + greeting + HORIZONTAL_LINE);
+    }
+    public static void bye() {
+        String bye_word = "Bye. Hope to see you again soon!";
+        System.out.println(HORIZONTAL_LINE + "\n" + bye_word + "\n" + HORIZONTAL_LINE);
+    }
+
+    public static void addTask(Task t) {
+        tasks.add(t);
+        taskCount = tasks.size();
+        System.out.println(HORIZONTAL_LINE + "\n"
+                + "Got it. I've added this task:\n"
+                + "  " +t.toString() + "\n"
+                + "Now you have "+ taskCount + " duke.tasks in the list\n"
+                + HORIZONTAL_LINE);
+
+    }
+    public static void listTasks() {
+        System.out.println(HORIZONTAL_LINE + "\n"
+                + "Here are the duke.tasks in your list:");
+        for (int i = 0; i< taskCount; i++) {
+            System.out.println((i+1)+"."+tasks.get(i).toString());
+        }
+        System.out.println(HORIZONTAL_LINE);
+    }
+
+    public static Boolean checkTaskNum() {
+        boolean isNumValid = true;
+        try{
+            int inputNum = Integer.parseInt(description);
+            if (inputNum > taskCount || taskCount == 0 || inputNum == 0) {
+                throw new DukeException();
+            }
+        }catch (NumberFormatException e) {
+            isNumValid = false;
+            System.out.println(HORIZONTAL_LINE+"\n"
+                    + "☹ OOPS!!! The task number of the one you have done is invalid!\n"
+                    +HORIZONTAL_LINE);
+        }catch (DukeException e) {
+            isNumValid = false;
+            System.out.println(HORIZONTAL_LINE+"\n"
+                    + "☹ OOPS!!! You did not set that task!\n"
+                    +HORIZONTAL_LINE);
+        }
+        return isNumValid;
+    }
+    public static void doneTask(int taskIndex) {
+        System.out.println(HORIZONTAL_LINE);
+        tasks.get(taskIndex - 1).markAsDone();
+        System.out.println("Nice! I've marked this task as done:");
+        System.out.println("  " + tasks.get(taskIndex - 1).toString());
+        System.out.println(HORIZONTAL_LINE);
+    }
+
+    public static void deleteTask(int taskIndex) {
+        System.out.println(HORIZONTAL_LINE + "\n"
+                + "Noted. I've removed this task:\n"
+                + tasks.get(taskIndex - 1).toString());
+        tasks.remove(taskIndex - 1);
+        taskCount = tasks.size();
+        System.out.println("Now you have "+ taskCount +" tasks in the list.\n"
+                + HORIZONTAL_LINE);
+    }
+
+
+
+    public static String processCommand() {
```

Comment:
This method may be confusing as it also starts with `process` but unlike the other methods it returns a String instead of being a void method.